.github/workflows/test-gha-eks.yml

```diff
@@ -0,0 +1,182 @@
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+
+env:
+
+  AWS_PROFILE: "infex"
+  AWS_REGION: "eu-west-2"
+    # /!\ always use one of the available test region https://github.com/camunda/infraex-common-config
+
+    # please keep those synced with tests.yml
+  TF_STATE_BUCKET: "tests-eks-tf-state-eu-central-1"
+  TF_STATE_BUCKET_REGION: "eu-central-1"
+
+  CREATE_DB: "${{ github.event.inputs.create_db || 'true' }}"
+
+jobs:
+  action-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+
+      - name: Install tooling using asdf
+        uses: asdf-vm/actions/install@05e0d2ed97b598bfce82fd30daf324ae0c4570e6 # v3
+
+      - name: Get Cluster Info
+        id: commit_info
+        run: |
+          if [[ -n "${{ inputs.cluster_name }}" ]]; then
+            cluster_name="${{ inputs.cluster_name }}"
+          else
+            cluster_name="cl-$(git rev-parse --short HEAD)"
+          fi
+          echo "cluster_name=$cluster_name" | tee -a "$GITHUB_OUTPUT"
+
+          if [[ -n "${{ inputs.db_username }}" ]]; then
+            db_username="${{ inputs.db_username }}"
+          else
+            db_username="user$(openssl rand -hex 4 | tr -d '/@" ')"
+          fi
+          echo "db_username=$db_username" | tee -a "$GITHUB_OUTPUT"
+
+          if [[ -n "${{ inputs.db_password }}" ]]; then
+            db_password="${{ inputs.db_password }}"
+          else
+            db_password="$(openssl rand -base64 12 | tr -d '/@" ')"
+          fi
+          echo "db_password=$db_password" | tee -a "$GITHUB_OUTPUT"
+
+          # Get the current commit hash for the modules revision
+          tf_modules_revision=$(git rev-parse HEAD)
+          echo "tf_modules_revision=$tf_modules_revision" | tee -a "$GITHUB_OUTPUT"
+
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@d1720f055e0635fd932a1d2a48f87a666a57906c # v3
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          exportEnv: false
+          secrets: |
+            secret/data/products/infrastructure-experience/ci/common AWS_ACCESS_KEY;
+            secret/data/products/infrastructure-experience/ci/common AWS_SECRET_KEY;
+
+      - name: Add profile credentials to ~/.aws/credentials
+        run: |
+          aws configure set aws_access_key_id ${{ steps.secrets.outputs.AWS_ACCESS_KEY }} --profile ${{ env.AWS_PROFILE }}
+          aws configure set aws_secret_access_key ${{ steps.secrets.outputs.AWS_SECRET_KEY }} --profile ${{ env.AWS_PROFILE }}
+          aws configure set region ${{ env.AWS_REGION }} --profile ${{ env.AWS_PROFILE }}
+
+      - name: Create EKS Cluster
+        timeout-minutes: 125
```

Comment:
since you ran it a couple of times, maybe we could lower it to a more realistic value.
Maybe 45 minutes, that already includes a bit of buffer time